source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -5,43 +5,55 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.eclipse.jface.viewers.ISelectionChangedListener;
-import org.eclipse.jface.viewers.SelectionChangedEvent;
+import org.eclipse.jface.fieldassist.ComboContentAdapter;
+import org.eclipse.jface.fieldassist.ContentProposal;
+import org.eclipse.jface.fieldassist.ContentProposalAdapter;
+import org.eclipse.jface.fieldassist.IContentProposal;
+import org.eclipse.jface.fieldassist.IContentProposalProvider;
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
 import org.eclipse.swt.events.SelectionAdapter;
 import org.eclipse.swt.events.SelectionEvent;
 import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Label;
 
-import com.microsoft.tfs.client.common.framework.command.ICommandExecutor;
 import com.microsoft.tfs.client.common.ui.Messages;
 import com.microsoft.tfs.client.common.ui.browser.BrowserFacade;
 import com.microsoft.tfs.client.common.ui.browser.BrowserFacade.LaunchMode;
 import com.microsoft.tfs.client.common.ui.controls.generic.BaseControl;
 import com.microsoft.tfs.client.common.ui.controls.generic.compatibility.link.CompatibilityLinkControl;
 import com.microsoft.tfs.client.common.ui.controls.generic.compatibility.link.CompatibilityLinkFactory;
-import com.microsoft.tfs.client.common.ui.framework.command.UICommandExecutorFactory;
-import com.microsoft.tfs.client.common.ui.framework.command.UICommandFinishedCallbackFactory;
+import com.microsoft.tfs.client.common.ui.dialogs.connect.ServerListDialog;
 import com.microsoft.tfs.client.common.ui.framework.helper.SWTUtil;
 import com.microsoft.tfs.client.common.ui.framework.layout.GridDataBuilder;
-import com.microsoft.tfs.core.util.ServerURIUtils;
+import com.microsoft.tfs.core.config.persistence.DefaultPersistenceStoreProvider;
 import com.microsoft.tfs.core.util.URIUtils;
+import com.microsoft.tfs.core.util.serverlist.ServerList;
+import com.microsoft.tfs.core.util.serverlist.ServerListConfigurationEntry;
+import com.microsoft.tfs.core.util.serverlist.ServerListManagerFactory;
+import com.microsoft.tfs.util.StringHelpers;
 import com.microsoft.tfs.util.listeners.SingleListenerFacade;
 
 public class ServerTypeSelectControl extends BaseControl {
     private static final Log log = LogFactory.getLog(ServerTypeSelectControl.class);
 
     private URI serverURI = null;
-    private boolean serverReadonly = false;
     private boolean ignoreServerChangeEvents = false;
-    private ICommandExecutor commandExecutor;
-    private ICommandExecutor noErrorDialogCommandExecutor;
-    private final ServerSelectControl serverControl;
+    private Combo serverCombo;
+    private ContentProposalAdapter serverComboAdapter;
```

Comment:
Why do we need serverComboAdapter as a class variable? It is only locally used in the single method setupServerCombo, isn't it?